src/FSharpLint.Core/Rules/Conventions/UnneededRecKeyword.fs

```diff
@@ -0,0 +1,44 @@
+﻿module FSharpLint.Rules.UnneededRecKeyword
+
+open System
+open FSharp.Compiler.Syntax
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.Rules
+open FSharpLint.Framework
+open FSharpLint.Framework.Suggestion
+
+let runner (args: AstNodeRuleParams) =
+    match args.AstNode, args.CheckInfo with
+    | AstNode.ModuleDeclaration (SynModuleDecl.Let (isRecursive, bindings, _)), Some checkInfo when isRecursive ->
+        match bindings with
+        | SynBinding (_, _, _, _, _, _, _, SynPat.LongIdent (longDotId, _, _, _, _, range), _, _, _, _) :: _ ->
+            let symbolUses = checkInfo.GetAllUsesOfAllSymbolsInFile()
+            let funcName = longDotId.Lid.Head.idText
+
+            let functionCalls =
+                symbolUses
+                |> Seq.filter (fun (symbol) -> symbol.Symbol.DisplayName = funcName)
```

Comment:
You also need to check if symbol is used inside function body. You can compare ranges for that.

Reply:
@webwarrior-ws isn't it better to add a new test first to cover this scenario?

Reply:
Yes, a test with function used outside function body some would be helpful

Reply:
@Mersho ok Mershad go ahead and add this, with 2 commits like we explained above